### DISCUSSION TITLE
BERT and spacy - is it possible to use BERT masked language model from spacy?

### DISCUSSION THREAD

**Discussion by ophiry** [explosion/spaCy#7542] [Help: Other Questions] [answered]
I know spacy integrates BERT for generating word \ document embeddings.
is it also use BERT for predicting missing \ masked words?

(using BERT as an external model is of course possible, but the integration seems non trivial - for example handling the different tokenization methods)

**Comment by polm** [comment, marked as answer]
Hey, sorry for the delayed reply on this. Short answer is no, Natural Language Generation type tasks are considered out of scope for spaCy. 

You could do something using it as a pipeline component. The English Transformer model is already based on RoBERTa, you can look at the source of [spacy-transformers](https://github.com/explosion/spacy-transformers) for more details on how the difference in tokenization is handled.
